### PR TITLE
Stats: Add Mobile Apps promo card via DotPager UI.

### PR DESCRIPTION
**Pull request by a8ck3n**
#### Proposed Changes

Adds the Jetpack mobile apps promo card alongside the Yoast SEO one using the `DotPager` component. The Yoast card is still the default.

Before:

APromoSEO

After:

NewPromoCards

#### Testing Instructions

1. Visit the Calypso Live link.
2. Visit the Stats page.
3. Scroll down and confirm the `DotPager` card is present.
4. Confirm the Yoast SEO card is the default. 
5. Confirm the mobile apps card is available on the 2nd page.
6. Confirm it behaves appropriately on smaller screen sizes.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #69230 and supersedes #70059.

Currently just updating the Stats page. The designs call for the same promo card on the Insights and Ads pages but it looks overpowering there with the older page layout. I'm thinking we should wait until we "grid" those layouts before adding this component in.

### CODE REVIEW

**Review by keoshi — DISMISSED** [review]
Hey, @a8ck3n — thanks for updating this. I noticed a few things happening:

- [x] In cases where the Yoast banner does not appear, we should not have the `DotPager` element and simply show the new banner.

image

- [x] The wrapper card should have a white background.

image

- [x] On mobile, the wrapper card should behave like all others: lose the `border-radius` and `border-left` and `-right`.

image

- [x] On mobile, the QR should change to the corresponding app store button as proposed in p1HpG7-iJR-p2. We can probably use the buttons in the footer of this page as reference: https://apps.wordpress.com/mobile/

Before | After
--|--
image | image

**Comment by a8ck3n** [comment]
Hey @keoshi, are you able to reproduce this?

> In cases where the Yoast banner does not appear, we should not have the `DotPager` element and simply show the new banner.

There isn't any logic in the code that manages when to show the Yoast banner. It should **always** be shown on the main Traffic page. My guess is that you encountered a JavaScript or network error during your testing that essentially broke the render.

If you can reproduce this, please let me know if there's anything listed in the console. In my testing, I see two console errors which shouldn't impact page rendering. Those are:

1. "Encountered two children with the same key, […]" — This is from the "?" mark help button in the top-right of the page.
2. "Invalid DOM property `stroke-width`." — This is from the SVG data and is being addressed [here](https://github.com/Automattic/wp-calypso/pull/70165).

Thanks!

**Comment by a8ck3n** [comment]
Hello @keoshi! Items 2 & 3 from the above list have been addressed. As noted, I'm unable to reproduce item 1. If you can review your console output and let me know what you are seeing (and if the issue persists), that would be great.

Regarding the fourth item, showing the correct app store badge on mobile, I'll add that to this [separate PR](https://github.com/Automattic/wp-calypso/pull/70165) since that is handled in the component and not as part of the wrapper/container.

Thanks!

**Comment by a8ck3n** [comment]
This is addressed in PR 70165 [here](https://github.com/Automattic/wp-calypso/pull/70165).

> On mobile, the QR should change to the corresponding app store button as proposed in p1HpG7-iJR-p2.

**Comment by keoshi** [comment]
> The first issue you noted, where the Yoast card is blank, is an odd one.

I think it might be just that I’m looking at Stats through a Jetpack site and the `PromoCardBlock` has some logic to check certain type sites and whether the banner is displayed or not:

https://github.com/Automattic/wp-calypso/blob/408c337c76209b2cb2c9334eccd1ef6e3762c9e4/client/blocks/promo-card-block/index.jsx#L42-L44

**Comment by dognose24** [comment]
> I think it might be just that I’m looking at Stats through a Jetpack site and the PromoCardBlock has some logic to check certain type sites and whether the banner is displayed or not:

I have also encountered this blank card, which resulted from the `jetpackNonAtomic` [checking](https://github.com/Automattic/wp-calypso/blob/408c337c76209b2cb2c9334eccd1ef6e3762c9e4/client/blocks/promo-card-block/index.jsx#L43-L45) of my local dev site.

截圖 2022-11-23 下午10 56 58

File: client/my-sites/stats/site.jsx
Comment (dognose24) [inline client/my-sites/stats/site.jsx]:
```suggestion
				<div className="stats__promo-container">
```

According to the [naming guide of BEM](https://getbem.com/naming/), could we have this container with a more comprehensive name and append the modifier `--traffic` when necessary?

File: client/my-sites/stats/site.jsx
Comment (dognose24) [inline client/my-sites/stats/site.jsx]:
```suggestion
					<DotPager className="stats__promo-pager">
```

As what is mentioned above for BEM.

File: client/my-sites/stats/site.jsx
Comment (dognose24) [inline client/my-sites/stats/site.jsx]:
```suggestion
						<div className="stats__promo-card">
```

As what is mentioned above for BEM.

File: client/my-sites/stats/site.jsx
Comment (a8ck3n) [reply client/my-sites/stats/site.jsx]: Thanks for the BEM link, Dognose. I reviewed that and the [Calypso CSS guidelines](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md) and I think I've got it right now. Let me know if you see anything else that isn't a best practice!

**Comment by a8ck3n** [comment]
- Added a check matching the one in the `PromoCardBlock` so that we can detect when we need the `DotPager` UI and when we don't.
- Cleaned up the class naming and CSS to better follow Calypso naming conventions.
- Tweaked the padding inside the `DotPager` to better match the original designs.

**Comment by a8ck3n** [comment]
I'm renaming the `JetpackMobilePromoCard` to `MobilePromoCard` now that it handles both Jetpack and Woo. See related PR [here](https://github.com/Automattic/wp-calypso/pull/70165). Therefore, we need to hold this PR and update it to use the updated component name once #70165 is merged.

**Comment by dognose24** [comment]
The [blocker of this PR](https://github.com/Automattic/wp-calypso/pull/70165) has been merged! cc @a8ck3n @keoshi

**Pull request closed without merging**
